src/codeu/chat/client/commandline/Chat.java

```diff
@@ -324,19 +344,174 @@ public void invoke(List<String> args) {
         }
       }
+    });
 
-      // Find the first conversation with the given name and return its context.
-      // If no conversation has the given name, this will return null.
-      private ConversationContext find(String title) {
-        for (final ConversationContext conversation : user.conversations()) {
-          if (title.equals(conversation.conversation.title)) {
-            return conversation;
+    // U-INT-ADD (add user interest)
+    //
+    // Add a command that will add a user to the current user's
+    // interests when the user enters "u-int-add" while on the user panel
+    //
+    panel.register("u-int-add", new Panel.Command() {
+      @Override
+      public void invoke(List<String> args) {
+        for(String token : args){
+          final String name = token;
+          if (name.length() > 0) {
+            final UserContext foundUser = findUser(context,name);
+            if (foundUser == null) {
+              System.out.format("ERROR: No user with name '%s'\n", name);
+            } else if (user.addUserInterest(name) == true) {
+                System.out.println("User \"" + name + "\" added to interests");
+            } else {
+                System.out.println("ERROR: User \"" + name + "\" already in interests");
+            }
+          } else {
+            System.out.println("ERROR: Missing <username>");
+          }
+        }
+      }
+    });
+
+    // C-INT-ADD (add conversation interest)
+    //
+    // A command that will add a conversation to the current user's
+    // interests map when the user enters "c-int-add"" while on the user panel
+    //
+    panel.register("c-int-add", new Panel.Command() {
+      @Override
+      public void invoke(List<String> args) {
+        for(String token : args){
+          final String title = token;
+          if (title.length() > 0) {
+            final ConversationContext conversation = findConvo(user,title);
+            if (conversation == null) {
+              System.out.format("ERROR: No conversation with name '%s'\n", title);
+            } else if (user.addConversationInterest(title) == true) {
+                System.out.println("Conversation \"" + title + "\" is added to interests");
+            } else {
+                System.out.println("ERROR: Conversation \"" + title + "\" is already in interests");
+            }
+          } else {
+            System.out.println("ERROR: Missing <title>");
+          }
+        }
+      }
+    });
+
+    // U-INT-REMOVE (Remove user interest)
+    //
+    // Add a command that will remove a user to the current user's
+    // interests when the user enters "u-int-remove" while on the user panel
+    //
+    panel.register("u-int-remove", new Panel.Command() {
+      @Override
+      public void invoke(List<String> args) {
+        for(String token : args){
+          final String name = token;
+          if (name.length() > 0) {
+            final UserContext foundUser = findUser(context,name);
```

Comment:
nit: same here,  and in other instances below. I'll stop leaving an individual comment for each one :)